Source/PluginProcessor.h

```diff
@@ -72,17 +74,39 @@ class SoomplerAudioProcessor  : public AudioProcessor, ChangeListener
 
     void setTransportStateListener(TransportStateListener*);
 
-    double getCurrentAudioPosition() const;
+    double getCurrentAudioPosition();
+
+    int64 getTotalLengthOfSample() const {
+        return transportSource.getTotalLength();
+    }
+
+    void updateTransportState();
 
     void processTransport(AudioBuffer<float>& buffer);
 
+    // percent of a sample length
+    void setSampleStartPosition(int64 sample);
+    void setSampleEndPosition(int64 sample);
+
+    void setVolume(double volume) {

```

Comment:
You need to try not to define functions in header files. Because if you'll want to change say volume parameter name then all code that depends on PluginProcessor.h should be recompiled. Think about it. In the future, if the project will be big enough that will slowdown the development process.

Reply:
ok, i leave this function this way, cos for some reason it didn't work when defined in cpp file.